src/pip/_internal/cache.py

```diff
@@ -77,6 +84,44 @@ def _get_cache_path_parts(self, link):
 
         return parts
 
+    def _get_cache_path_parts(self, link):
+        # type: (Link) -> List[str]
+        """Get parts of part that must be os.path.joined with cache_dir
+        """
+
+        # We want to generate an url to use as our cache key, we don't want to
+        # just re-use the URL because it might have other items in the fragment
+        # and we don't care about those.
+        key_parts = {"url": link.url_without_fragment}
+        if link.hash_name is not None and link.hash is not None:
+            key_parts[link.hash_name] = link.hash
+        if link.subdirectory_fragment:
+            key_parts["subdirectory"] = link.subdirectory_fragment
+
+        # Include interpreter name, major and minor version in cache key
+        # to cope with ill-behaved sdists that build a different wheel
+        # depending on the python version their setup.py is being run on,
+        # and don't encode the difference in compatibility tags.
+        # https://github.com/pypa/pip/issues/7296
+        key_parts["interpreter"] = "{}-{}.{}".format(
```

Comment:
Can we split this into `interpreter_name` and `interpreter_version` and for `interpreter_version` use `pep425tags.get_impl_ver()`? We would also want to make a separate issue to switch it to `packaging.tags.interpreter_version()` when we integrate the new version.

Reply:
@chrahunt AFAICS `packaging.tags.interpreter_version()` does not exist, nor `packaging.tags.interpreter_name()` on `packaging` master branch.
Perhaps `list(packaging.tags.sys_tags())[0]` is better for both interpreter name and version?

Reply:
Sorry, it would be in pypa/packaging#231.

Reply:
Ok, thanks. I added aliases for `interpreter_name` and `interpreter_version` in `pep425tags.py` so it's obvious what to do when it's replaced with `packaging.tags`.